extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.ViewPageTarget.js

```diff
@@ -92,20 +92,54 @@ ve.init.wikia.ViewPageTarget.static.actionsToolbarConfig = [
 ve.init.wikia.ViewPageTarget.prototype.setSurface = function ( surface ) {
 	// Parent method
 	ve.init.mw.ViewPageTarget.prototype.setSurface.call( this, surface );
-	this.setupLicense( surface );
+
+	if ( !this.isSaving ) {
+		if (!this.loginWidget) {
+			this.loginWidget = new ve.ui.WikiaLoginWidget();
+			this.loginWidget.setupAnonWarning(this.getToolbar());
+		}
+		
+		this.setupLicense( surface );
+	}
+};
+
+/**
+ * @inheritdoc
+ */
+ve.init.wikia.ViewPageTarget.prototype.onSave = function (
+	html, categoriesHtml, newid, isRedirect, displayTitle, lastModified, contentSub
+) {
+	this.tearDownLicense();
+	this.tearDownAnonWarning();
+	this.isSaving = true;
+
+	// Parent method
+	return ve.init.mw.ViewPageTarget.prototype.onSave.call(this, html, categoriesHtml, newid, isRedirect, displayTitle, lastModified, contentSub);
 };
 
 /**
  * @inheritdoc
  */
 ve.init.wikia.ViewPageTarget.prototype.tearDownSurface = function ( noAnimate ) {
+	this.tearDownLicense();
+	this.tearDownAnonWarning();
+
+	// Parent method
+	return ve.init.mw.ViewPageTarget.prototype.tearDownSurface.call( this, noAnimate );
+};
+
+ve.init.wikia.ViewPageTarget.prototype.tearDownLicense = function() {
 	if ( this.$license ) {
 		this.$license.remove();
 		this.$license = null;
 	}
+};
 
-	// Parent method
-	return ve.init.mw.ViewPageTarget.prototype.tearDownSurface.call( this, noAnimate );
+ve.init.wikia.ViewPageTarget.prototype.tearDownAnonWarning = function() {
+	if ( this.loginWidget ) {
+		this.loginWidget.removeAnonWarning();
+		this.loginWidget = null;
+	}
 };
 
 //ve.init.wikia.ViewPageTarget.static.surfaceCommands.push( 'wikiaSourceMode' );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this commented out code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ACommentedCode)